use-cases/call-center/src/test/java/org/acme/callcenter/solver/SolverServiceTest.java

```diff
@@ -96,6 +105,63 @@ void removeCall() {
         assertThat(call.getId()).isEqualTo(call2.getId());
     }
 
+    @Test
+    @Timeout(60)
+    void removeCallWithInfeasibleInitialSolution() {
+        // Invalid initial solution
+        CallCenter inputProblem = dataGenerator.generateCallCenter();
+        Call call1 = new Call(1L, "123-456-7891", Skill.ENGLISH, Skill.CAR_INSURANCE);
+        Call call2 = new Call(2L, "123-456-7892", Skill.ENGLISH, Skill.CAR_INSURANCE);
+        inputProblem.getCalls().add(call1);
+        inputProblem.getCalls().add(call2);
+        inputProblem.getAgents().get(0).setNextCall(call2);
+        inputProblem.getAgents().get(1).setNextCall(call1);
+        call1.setAgent(inputProblem.getAgents().get(1));
+        call1.setPreviousCallOrAgent(inputProblem.getAgents().get(1));
+        call1.setEstimatedWaiting(Duration.ofSeconds(1));
+        call2.setAgent(inputProblem.getAgents().get(0));
+        call2.setPreviousCallOrAgent(inputProblem.getAgents().get(0));
+        call2.setEstimatedWaiting(Duration.ofSeconds(1));
+        inputProblem.setScore(HardSoftScore.ofUninitialized(0, -1, 0));
+
+        // We have an initial solution, then we run only LocalSearchPhaseConfig
+        SolverConfig solverConfig = new SolverConfig()
+                .withSolutionClass(CallCenter.class)
+                .withEntityClasses(Call.class, PreviousCallOrAgent.class)
+                .withConstraintProviderClass(CallCenterConstraintsProvider.class)
+                .withPhases(new LocalSearchPhaseConfig())
+                .withTerminationConfig(new TerminationConfig().withSpentLimit(Duration.ofSeconds(30L)));
+
+        // Create a custom manager
+        try (SolverManager<CallCenter, Long> solverManager2 = SolverManager.create(solverConfig, new SolverManagerConfig())) {
```

Comment:
Numbering variables harms readability. While I am technically guilty of doing that as well, I see some sense in breaking the rule for test data (`call1`, `call2`), provided it does not have any specific purpose in the test.

May I suggest renaming the variable to `localSearchSolverManager` ?